### PR TITLE
MEN-6142: fix: Patch `mender-artifact` git recipe for golang 1.14

### PR DESCRIPTION
**Pull request by lluiscampos**
The incompatibilities are in the vendored dependencies of
`mender-artifact`. Two patches revert specific upstream changes that
introduced a dependency on golang newer than 1.14 and the two others are
direct patches on the source code.

### CODE REVIEW

**Review by oleorhagen — COMMENTED** [review]
I don't see anything jumping out at me.

Will have a look at the pipes

**Comment by lluiscampos** [comment]
Related CI fix at:
* https://github.com/mendersoftware/mender-qa/pull/666

**Comment by lluiscampos** [comment]
Build @ [Pipeline](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/765819947)

File: meta-mender-core/recipes-mender/mender-artifact/patches/0004-golang-1.14-compat-patch-hashicorp-go-secure-stdlib-.patch [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
++	MinInt16  = -1 << 15
++	MaxInt32  = 1<<31 - 1
++	MinInt32  = -1 << 31
++	MaxInt64  = 1<<63 - 1
++	MinInt64  = -1 << 63
++	MaxUint   = 1<<intSize - 1
++	MaxUint8  = 1<<8 - 1
++	MaxUint16 = 1<<16 - 1
++	MaxUint32 = 1<<32 - 1
++	MaxUint64 = 1<<64 - 1
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-artifact/patches/0004-golang-1.14-compat-patch-hashicorp-go-secure-stdlib-.patch:51]: 👍

File: meta-mender-core/recipes-mender/mender-client/patches/0001-golang-1.14-compat-Amend-unit-tests-with-a-custom-te.patch [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
++	"os"
++	"testing"
++)
++
++// Setenv calls os.Setenv(key, value) returns a function to
++// restore the environment variable to its original value
++//
++// Original source:
++// https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/testing/testing.go;l=1265
++func Setenv(t *testing.T, key, value string) func() {
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/patches/0001-golang-1.14-compat-Amend-unit-tests-with-a-custom-te.patch:94]: Nice!

**Review by kacf — APPROVED** [review]

**Review by oleorhagen — APPROVED** [review]

**Pull request merged**